### DISCUSSION TITLE
Accessing all App Components from SDK

### DISCUSSION THREAD

**Discussion by JayP718** [Budibase/budibase#8758] [Help] [answered]
Hi,

I'm trying to build a component which will allow users to create walkthroughs for their apps. However, it requires two custom components to work so one component needs to be able to read from another in a sense. They work as a pair.

Right now I'm doing that by passing it through the dom and having the other component scan the dom to read details from the other components. This works great with the exception of modals since the dom changes and "adds" new components to the screen (conditional visibility might also pose an issue). I can cater for this with a somewhat hacky solution but before doing that I was wondering if there was a way to get all components of an app through the SDK. The component store seems to only be filled when in the builder and not . The screen store only has components on the current screen and doesn't change when a modal is opened. I don't see an action to retrieve other screens but I want to get the components of the entire app to see if that component exists.

Hopefully I'm doing a decent job of explaining.

Appreciate any help.

**Comment by melohagan** [comment]
Hey @JayP718 

I'm not certain, but I believe each component is fairly encapsulated, in that it isn't really aware of other components. 
There is a way to provide context to child components. Might be worth having a read through this:
https://docs.budibase.com/docs/custom-component#example-context-provider

**Reply by JayP718** [reply]
Thanks Mel, I'll have a look

**Comment by JayP718** [comment, marked as answer]
For others who come across this it's Possible using Screen api call. Check out my tour plugin if you want to see how.

**Reply by melohagan** [reply]
Nice one - good job!
